### ISSUE TITLE
TypeError: DataLoaderIterWrapper object is not an iterator ?

### ISSUE THREAD

**Issue by LightToYang** [davidtvs/pytorch-lr-finder#33]
```
class DataLoaderIterWrapper(object):
     def __init__(self, data_loader, auto_reset=True):
        self.data_loader = data_loader
        self.auto_reset = auto_reset
        self._iterator = iter(data_loader)

    def __next__(self):
        # Get a new set of inputs and labels
        try:
            # inputs, labels, *_ = next(self._iterator)
            inputs, labels = next(self._iterator)
        except StopIteration:
            if not self.auto_reset:
                raise
            self._iterator = iter(self.data_loader)
            # inputs, labels, *_ = next(self._iterator)
            inputs, labels = next(self._iterator)

        return inputs, labels
```

**Comment by NaleRaphael** [comment]
Hi @LightToYang, could you post the traceback of this error and some description of the dataset you are using?

**Comment by LightToYang** [comment]
```
Traceback (most recent call last):
  File "train_find_lr.py", line 185, in <module>
    main()
  File "train_find_lr.py", line 180, in main
    lr_finder.range_test(train_loader, end_lr=100, num_iter=3000)
  File "/home/young/workspace/ImageNetFramework/tricks/lr_finder.py", line 195, in range_test
    loss = self._train_batch(iter_wrapper, accumulation_steps)
  File "/home/young/workspace/ImageNetFramework/tricks/lr_finder.py", line 243, in _train_batch
    inputs, labels = next(iter_wrapper)
TypeError: DataLoader object is not an iterator
```
I use python2 rather than python3

**Comment by NaleRaphael** [comment]
Since we have dropped support for Py2k currently, you could try to install this package with version 1.0 for it if you are not planning to migrate to Py3k.

Or if you are going to customize this package (according to the traceback you given), you could simply add `next = __next__` in `class DataLoaderIterWrapper` to make it compatible with Py2k.
See also:
https://github.com/davidtvs/pytorch-lr-finder/blob/a1379836bc4ab7863713b7ee525a64c1b6d43c0c/torch_lr_finder/lr_finder.py#L436-L461

**Comment by davidtvs** [comment]
As @NaleRaphael said we dropped support for Py2k recently. However, the last release was left in a broken state for Py2k so it's only fair that it gets fixed. I released version 0.1.5 which fixes the extended iterable unpacking meaning that you shouldn't have to comment-out those lines and further change the code to make it work.

Give it a try, it's currently the latest version on PyPI so you can simply do `pip install torch-lr-finder`.

I also created a branch with the latest code that works on Py2k: [torch_lr_finder-v0.1](https://github.com/davidtvs/pytorch-lr-finder/tree/torch_lr_finder-v0.1).

Let us know if you have any problems with the latest version

**Comment by davidtvs** [comment]
Closing this, feel free to reopen if you find a problem related to this
